### PR TITLE
Customizable Banners!

### PR DESCRIPTION
**Pull request by MatheusIsNotAvailable**
[FEATURE] [WIKI]

## About the PR 
Adds Banners which can be made by using cloth on a wall, the banner inherits the cloth's material, the banner can then be colored by using a crayon on it, after it has been colored you can draw various patterns or letters on it, there is no limit to how many you can draw they are the same color as the crayon you used to draw them too, you can clear it of it's color and drawings by click dragging it, and you can remove it completely by using a wire cutter or similar on it which will give you the cloth you used back. Also makes a small change to rainbow crayons so they work as you expect them to when used in a banner.
## Why's this needed? 
It is cool, lets players customize places in a cool way
Banners!

## Changelog

```changelog
(u)Arthur Holiday
(*)Added Customizable banners, use cloth on a wall to make one, color and draw on it with crayons.
```

### CODE REVIEW

**Comment by Gannets** [comment]
Can't speak to the code quality but this is a really nice idea, sprites look good too. 👍

File: code/obj/decal/poster.dm
Comment (PrimeNumb) [inline code/obj/decal/poster.dm:990]:
```suggestion
	var/colored = FALSE
```

File: code/obj/decal/poster.dm
Comment (PrimeNumb) [inline code/obj/decal/poster.dm:1000]:
```suggestion
		src.colored = FALSE
```

File: code/obj/decal/poster.dm
Comment (PrimeNumb) [inline code/obj/decal/poster.dm:1012]:
```suggestion
			if(src.colored == TRUE)
```

File: code/obj/decal/poster.dm
Comment (PrimeNumb) [inline code/obj/decal/poster.dm:1024]:
```suggestion
				src.colored = TRUE
```

**Review by PrimeNumb — CHANGES_REQUESTED** [review]

File: code/obj/decal/poster.dm
Comment (PrimeNumb) [inline code/obj/decal/poster.dm:988]: looks like `popup_win` is never used?

File: code/obj/decal/poster.dm
Comment (PrimeNumb) [inline code/obj/decal/poster.dm:1022]:
```suggestion
			if(src.colored == FALSE)
```

**Review by PrimeNumb — CHANGES_REQUESTED** [review]

File: code/modules/materials/Mat_RawMaterials.dm
Comment (ZeWaka) [inline code/modules/materials/Mat_RawMaterials.dm:181]:
```suggestion
			if(locate(/obj/decal/poster/banner, A))
```
consteienycy

File: code/modules/materials/Mat_RawMaterials.dm
Comment (ZeWaka) [inline code/modules/materials/Mat_RawMaterials.dm:186]: weirdly worded please be more clear

File: code/modules/materials/Mat_RawMaterials.dm
Comment (ZeWaka) [inline code/modules/materials/Mat_RawMaterials.dm:188]: poorly worded

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:984]: if it's proper why do you have it lowercase

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:989]: doubtful

File: code/obj/item/pens_writing_etc.dm
Comment (ZeWaka) [inline code/obj/item/pens_writing_etc.dm:393]: why are you doing afterattack and not attackby on the actual banner object - this block of code does not make sense to me

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:996]: clear what?

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:998]: indent your blocks

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1008]:
```suggestion
```
uselsss

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1005]: umm why is there a blank argument what's going on here!!!

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1004]:
```suggestion
		. = ..()
```

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1012]:
```suggestion
			if(src.colored)
```
uncnessary

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1013]: null? i like titles on my windows don't you?

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1022]:
```suggestion
			else
```

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1041]: please indent your blocks after a control statement

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1039]: please be consistent in your usage of spaces for arguments
```suggestion
		else if(tgui_alert(usr, "Are you sure you want to clear the banner?", "Confirmation", list("Yes", "No")) == "Yes")
```

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1027]:
```suggestion
		if(istool(W,TOOL_SNIPPING | TOOL_CUTTING | TOOL_SAWING))
```
AUGH

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1031]: spaces after multipronged control statements plase

File: code/obj/decal/poster.dm
Comment (ZeWaka) [inline code/obj/decal/poster.dm:1018]: consider using overlay manager

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Review by Carbadox — COMMENTED** [review]
Tested locally and it works as intended.

My own suugestions would be to make that splashing a banner with acetone would reset it back to the original clean state and maybe also make acid destroy it. 

All in all though, very cool.

**Comment by MatheusIsNotAvailable** [comment]
Github mobile hid a bunch of the earlier reviews so I missed some stuff, gonna do the rest today probably, sorry for requesting a review when it isn't done yet

**Review by virvatuli-gh — COMMENTED** [review]
Looks good 2 me but I'll wait for Zewaka to check in before merge

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**
